### PR TITLE
fix: add validation to search by mail or t and show all external tran…

### PR DESCRIPTION
**Pull request by yoavrozner**
…sffers in the sidenav

### CODE REVIEW

File: src/components/inputs/BaseAutocomplete.vue [hunk 1]
```diff
@@ -19,7 +19,7 @@
     :class="{ 'disableInput' : disabled }"
   >
     <template v-slot:no-data>
-      <p id="no-resault">{{ $t(`autocomplete.${noResult || 'NoResult'}`)}}</p>
+      <p id="no-resault">{{ $t(`autocomplete.${ validationFailedMsg ? validationFailedMsg : noResult || 'NoResult'}`)}}</p>
```
Comment (nettashviro) [inline src/components/inputs/BaseAutocomplete.vue:22]: typo
```suggestion
      <p id="no-result">{{ $t(`autocomplete.${ validationFailedMsg ? validationFailedMsg : noResult || 'NoResult'}`)}}</p>
```

File: src/i18n/he.json [hunk 2]
```diff
@@ -28,7 +28,9 @@
     "NoResult": "לא נמצאו תוצאות",
     "NoResultAdvancedSearch": "לא נמצאו תוצאות. נסו חיפוש מתקדם.",
     "Drive": "חיפוש בדרייב...",
-    "Users": "הוסיפו אנשים..."
+    "Users": "הוסיפו אנשים...",
```
Comment (nettashviro) [inline src/i18n/he.json:31]: add in english version

File: src/components/popups/share-popup/tabs/Share.vue [hunk 3]
```diff
@@ -182,6 +184,9 @@ export default {
       this.selectedUsers = [];
       this.$emit("share", shareObject);
     },
+    advancedSearchValidation() {
+      return validationAdvancedSearchFactory(AdvancedSearchToEnum(this.advancedSearchSelection));
```
Comment (nettashviro) [inline src/components/popups/share-popup/tabs/Share.vue:188]: very nice! tell to alexa that you have won

File: src/utils/advancedSearchValidation.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { AdvancedSearchEnum } from './advancedSearchEnum';
+
+export function validationAdvancedSearchFactory(advancedSearchEnum) {
+    switch (advancedSearchEnum) {
+        case AdvancedSearchEnum.FindByMail:
+            return () => null;
```
Comment (nettashviro) [inline src/utils/advancedSearchValidation.js:6]: why this and the default case if they has the same results?

**Review by nettashviro — CHANGES_REQUESTED** [review]
very nice implementation!

File: src/components/popups/share-popup/tabs/Share.vue (hunk 3 above)
Comment (yoavrozner) [reply src/components/popups/share-popup/tabs/Share.vue:188]: Hahahaha :100:

File: src/utils/advancedSearchValidation.js (hunk 4 above)
Comment (yoavrozner) [reply src/utils/advancedSearchValidation.js:6]: Because it might change soon to a different validation function. Do you think I should remove it?

Comment (nettashviro) [reply src/utils/advancedSearchValidation.js:6]: yep

**Review by nettashviro — APPROVED** [review]

**Pull request merged**